### ISSUE TITLE
[Performance] Softmax Related Kernel Performance is not good

### ISSUE THREAD

**Issue by Stonepia** [intel/intel-xpu-backend-for-triton#287]
# Issue Description
We found that the softmax-related ops could not reach a competitive performance compared with CUDA. When testing the huggingface suite, the following models' performance is affected by the slow triton kernels. 

|                               | xpu time  (seconds) | CUDA time (seconds) |
|-------------------------------|---------------------|---------------------|
| BlenderbotSmallForCausalLM    | 0.035559            | 0.012942            |
| PLBartForCausalLM             | 0.007361            | 0.005116            |
| OPTForCausalLM                | 0.007202            | 0.004343            |
| GPT2ForSequenceClassification | 0.004554            | 0.003706            |
| DebertaV2ForQuestionAnswering | 0.000685            | 0.000439            |

# Reproducer
[reproducer.zip](https://github.com/intel/intel-xpu-backend-for-triton/files/13960190/reproducer.zip)

The attachment has the following structure.

- the `{model}_output_code_{device}.py`  contains the triton kernels. You could directly run them on the corresponding device. The result in the above table is printed by running these files.
- The `{model}_{device}.py` contains the aten ops, you could refer to that for understanding of the original logic. 

```Bash
.
├── BlenderbotSmallForCausalLM
│   ├── BlenderbotSmallForCausalLM_cuda.py
│   ├── BlenderbotSmallForCausalLM_output_code_cuda.py
│   └── BlenderbotSmallForCausalLM_output_code_xpu.py
├── DebertaV2ForQuestionAnswering
│   ├── DebertaV2ForQuestionAnswering_cuda.py
│   ├── DebertaV2ForQuestionAnswering_output_code_cuda.py
│   ├── DebertaV2ForQuestionAnswering_output_code_xpu.py
│   └── DebertaV2ForQuestionAnswering_xpu.py
├── OPTForCausualLM
│   ├── OPTForCausalLM_cuda.py
│   ├── OPTForCausalLM_output_code_cuda.py
│   ├── OPTForCausualLM_output_code_xpu.py
│   └── OPTForCausualLM_xpu.py
└── PLBartForCausualLM
    ├── PLBartForCasualLM_cuda.py
    ├── PLBartForCausalLM_output_code_cuda.py
    ├── PLBartForCausalLM_output_code_xpu.py
    └── PLBartForCausalLM_xpu.py 
```

Note that you may encounter issue when running `{model}_xpu.py`, you could change the `main()` like this:

```Python
if __name__ == '__main__':
    from torch._dynamo.repro.after_aot import run_repro
    from torch._dynamo.debug_utils import  InputReader
    input_reader = InputReader()
    load_args(input_reader)
    #  eager mode
    Repro()(*input_reader.args)
    #  inductor mode
    compiled_mod = torch.compile(mod)
    compiled_mod(*input_reader.args)
```

The `output_code.py` could be generated by running `TORCH_COMPILE_DEBUG=1 python xxx.py`, you could find it in the debug folder found in the output log. 

# System Info
XPU:
pytorch (internal): v2.1.0 44d1ed87a2675d0764e32b8839936371da8ea3a8 
ipex (internal): de704aec53780e6cb6fdf3db6db0485c9136bdab
triton: fb2b2a5d743c211c5b387d14ef789ce01c9ef601  (Tue Dec 26 Fix E2E doc transformers installation command (#175))

CUDA:
pytorch: v2.1.0 https://github.com/chuanqi129/pytorch/tree/chuanqiw/v2.1.0  4b6959d4a91fd39764b61a588122b54dd5e38522
triton: '2.1.0+e6216047b8' under pytorch folder, run `make triton`

**Comment by LiyangLingIntel** [comment]
@Stonepia Can you help check if this is still an issue on triton 3.0 version? If so, we can arrange effort on this issue.

**Comment by vlad-penkin** [comment]
Duplicates:
- #772 
- #773
